src/operators.jl

```diff
@@ -277,7 +277,8 @@ Base.sum(j::JuMPDict)  = sum(values(j.tupledict))
 Base.sum(j::JuMPArray{Variable}) = AffExpr(vec(j.innerArray), ones(length(j.innerArray)), 0.0)
 Base.sum(j::JuMPDict{Variable})  = AffExpr(collect(values(j.tupledict)), ones(length(j.tupledict)), 0.0)
 Base.sum(j::Array{Variable}) = AffExpr(vec(j), ones(length(j)), 0.0)
-function Base.sum{T<:GenericAffExpr}(affs::Array{T})
+Base.sum(j::AbstractArray{Variable}) = sum([j[i] for i in eachindex(j)]) # to handle non-one-indexed arrays.
```

Comment:
I tried that first, but `AffExpr` is a type alias of `GenericAffExpr{Float64,Variable}`, so that directly calls the inner constructor of `AffExpr`, which expects precisely a `Vector{Variable}`. I tried adding an inner constructor to `GenericAffExpr` that takes `AbstractArray`s (+ additional outer constructors), but ran into some more issues that I don't remember anymore. Basically, it was going to be a pretty big change, whereas `sum` seemed to be the only current case where handling `AbstractVector`s was needed when constructing `AffExpr`s.